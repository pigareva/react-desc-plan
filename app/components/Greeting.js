import React, { Component } from 'react';

export default class Greeting extends Component {
  render() {
    const hh = this.state.time / 60;
    switch (true) {
      case (hh >= 8 && hh <= 10):
        return <h2> Good Morning! </h2>;
      case (hh >= 12 && hh <= 14):
        return <h2> Bon appetite! </h2>;
      case (hh >= 17 && hh <= 20):
        return <h2> Bye bye! </h2>;
      default:
        return null;
    }
  }
}
