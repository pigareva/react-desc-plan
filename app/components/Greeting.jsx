import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class Greeting extends Component {
  render() {
    const hh = this.props.time / 60;
    switch (true) {
      case (hh >= 8 && hh <= 10):
        return <div className="greeting"> Good morning! </div>;
      case (hh >= 12 && hh <= 14):
        return <div className="greeting"> Bon appetite! </div>;
      case (hh >= 17 && hh <= 20):
        return <div className="greeting"> Bye bye! </div>;
      default:
        return null;
    }
  }
}

Greeting.propTypes = {
  time: PropTypes.number.isRequired,
};
