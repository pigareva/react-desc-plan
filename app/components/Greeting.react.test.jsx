import React from 'react';
import renderer from 'react-test-renderer';
import Greeting from './Greeting';

test('Greeting changes the text depending on time', () => {
  const componentMorning = renderer.create(<Greeting time={485} />);
  let tree = componentMorning.toJSON();
  expect(tree).toMatchSnapshot();

  const componentLunch = renderer.create(<Greeting time={780} />);
  tree = componentLunch.toJSON();
  expect(tree).toMatchSnapshot();

  const componentByeBye = renderer.create(<Greeting time={1080} />);
  tree = componentByeBye.toJSON();
  expect(tree).toMatchSnapshot();
});
