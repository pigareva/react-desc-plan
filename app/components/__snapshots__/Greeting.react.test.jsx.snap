// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting changes the text depending on time 1`] = `
<div
  className="greeting"
>
   Good morning! 
</div>
`;

exports[`Greeting changes the text depending on time 2`] = `
<div
  className="greeting"
>
   Bon appetite! 
</div>
`;

exports[`Greeting changes the text depending on time 3`] = `
<div
  className="greeting"
>
   Bye bye! 
</div>
`;
